libs/agno/agno/agent/agent.py

```diff
@@ -1044,9 +1046,15 @@ def run(
         messages: Optional[Sequence[Union[Dict, Message]]] = None,
         stream_intermediate_steps: bool = False,
         retries: Optional[int] = None,
+        knowledge_filters: Optional[Dict[str, Any]] = None,
         **kwargs: Any,
     ) -> Union[RunResponse, Iterator[RunResponse]]:
         """Run the Agent and return the response."""
+        if knowledge_filters:
+            if self.context is None:
+                self.context = {}
+        self.context["knowledge_filters"] = knowledge_filters
+        log_debug(f"Added knowledge filters to context: {knowledge_filters}")
```

Comment:
1. Any reason we don't set `context` to `{}` on class initialization? it still is a declarative way to say the context is empty, keeps the initialization tasks on initialization, and will save us from doing this in multiple places potentially
2. The logic should be in the if block I think?
```python
if knowledge_filters:
    self.context = {"knowledge_filters" = knowledge_filters}
    log_debug(f"Added knowledge filters to context: {knowledge_filters}")
```

Reply:
Yep, this whole bit is actually not required. pushed a change